File: Homework/Mary_Antonyan/C++/30_03_2017/Chess/pawn.cpp
#include "pawn.hpp"
#include "board.hpp"
#include <iostream>

//Constructor
Chess::Pawn::Pawn(const Chess::Figure::Chessman figure, 
                    const Chess::Figure::Color color, 
                    const int row, const int coloumn) 
    : Figure(figure, color, row, coloumn)
{}

//Destructor
Chess::Pawn::~Pawn() {}

//Method override
void Chess::Pawn::step(const int row, const int coloumn) {
    if (0 <= row && 8 > row && 0 <= coloumn && 8 > row) { 
        if ((((mColor == Chess::Figure::Color::white && mRow == 1) || 
           (mColor == Chess::Figure::Color::black && mRow == 6)) && row == mRow + 2) || 
           (row == mRow + 1 && coloumn == mColoumn && 
            Chess::board[row][coloumn] == nullptr) || 
            (((coloumn == mColoumn + 1 || coloumn == mColoumn -1) && 
             Chess::board[row][coloumn]->getColor() != mColor && 
             Chess::board[row][coloumn] != nullptr))) {
            std::cout << "Move to [" << row << " , " << coloumn << "]\n";
            int tmpRow = mRow;
            int tmpCol = mColoumn;
            Chess::board[row][coloumn] = new Pawn(mFigure, mColor, row, coloumn);
            Chess::board[tmpRow][tmpCol] = nullptr;
        } else {
            std::cout << "Can't move to [" << row << " , " << coloumn << "]\n";
        }
    } else {
        std::cout << "Out of board" << std::endl;
    }
}

File: Homework/Vrezh/java/chess/chess/Pawn.java
package chess;

public class Pawn extends Figure {
    private static int stepCount;

    public Pawn(String n, char column, int row, Color c) {
        super(n, column, row, c);
        stepCount = 0;
    }


    public boolean canGo(char c, int r) {
        int column = (int)this.column - 96;
        int col = (int)c - 96;
        System.out.println(col);
        
        System.out.println(r);

        System.out.println(column);
        System.out.println(row);
        if ((8 > r && color == Style.Color.BLACK) || (0 <= r && color == Style.Color.WHITE) && (column - col == 0)) {
            if (0 == stepCount && ((color == Style.Color.WHITE && ((r - row) == 2 || (r - row) == 1)) || (color == Style.Color.BLACK && ((row - r) == 2 || (row - r) == 1)))) { 
                ++stepCount;
                return true;
            } else if (1 == stepCount && ((color == Style.Color.WHITE && (col - column) <= 1) || (color == Style.Color.BLACK && (column - col) <= 1))) {
            return true;
            }
        }
        return false;
    }
    
} 

